File: src/cn/vove7/plugin/rest/filetype/RestHighlighterFactory.java
package cn.vove7.plugin.rest.filetype;

import com.intellij.openapi.fileTypes.SyntaxHighlighter;
import com.intellij.openapi.fileTypes.SyntaxHighlighterFactory;
import com.intellij.openapi.project.Project;
import com.intellij.openapi.vfs.VirtualFile;
import org.jetbrains.annotations.NotNull;

/**
 *
 */
public class RestHighlighterFactory extends SyntaxHighlighterFactory {

    private final RestHighlighter restHighlighter = new RestHighlighter();

    @NotNull
    @Override
    public SyntaxHighlighter getSyntaxHighlighter(Project project, VirtualFile virtualFile) {
        return restHighlighter;
    }
}

File: src/main/kotlin/cn/vove7/plugin/rest/filetype/RestHighlighterFactory.kt
package cn.vove7.plugin.rest.filetype

import com.intellij.openapi.fileTypes.SyntaxHighlighterFactory
import cn.vove7.plugin.rest.filetype.RestHighlighter
import com.intellij.openapi.vfs.VirtualFile
import com.intellij.openapi.fileTypes.SyntaxHighlighter
import com.intellij.openapi.project.Project

/**
 *
 */
class RestHighlighterFactory : SyntaxHighlighterFactory() {
    private val restHighlighter = RestHighlighter()
    override fun getSyntaxHighlighter(project: Project?, virtualFile: VirtualFile?): SyntaxHighlighter {
        return restHighlighter
    }
}